src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java

```diff
@@ -289,6 +316,21 @@ default MessageAction reference(@Nonnull Message message)
     @CheckReturnValue
     MessageAction mentionRepliedUser(boolean mention);
 
+    /**
+     * Whether to throw a exception if the referenced message was deleted, when replying to a message.
+     * <br>This only matters in combination with {@link #reference(Message)} and {@link #referenceById(long)}!
+     *
+     * <p>This is false by default but can be configured using {@link #setDefaultFailIfNotExists(boolean)}!
+     *
+     * @param  fail
+     *         True, to throw a exception if the referenced message was deleted
+     *
+     * @return Updated MessageAction for chaining convenience
+     */
+    @Nonnull
+    @CheckReturnValue
+    MessageAction failIfNotExists(boolean fail);
```

Comment:
What about ``failOnDeletedReply`` or ``failOnInvalidReply``? And if so, should all methods and fields be renamed accordingly?

Reply:
The latter sounds better since it will probably also fail if you try to reference a message from another channel. If you change it then yes please change the fields too so it's consistent. Also, you should reference this from the various reply/reference methods as well to improve discoverability.